ambry-frontend/src/main/java/com/github/ambry/frontend/GetAccountsHandler.java

```diff
@@ -187,5 +195,25 @@ private Container getContainer() throws RestServiceException {
       }
       return container;
     }
+
+    /**
+     * @return requested dataset.
+     * @throws RestServiceException
+     */
+    private Dataset getDataset() throws RestServiceException {
+      String accountName = RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.TARGET_ACCOUNT_NAME, true);
+      String containerName = RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.TARGET_CONTAINER_NAME, true);
+      String datasetName = RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.TARGET_DATASET_NAME, true);
+      try {
+        Container container = accountService.getContainerByName(accountName, containerName);
+        short accountId = container.getParentAccountId();
+        short containerId = container.getId();
+        return accountService.getDataset(accountId, containerId, accountName, containerName, datasetName);
```

Comment:
why not just accountService.getDataset(accountName, containerName, datasetName)?